src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -623,14 +829,30 @@ private static class Log
                 new EventId(2, "RequestBodyInvalidDataException"),
                 "Reading the request body failed with an InvalidDataException.");
 
-            public static void RequestBodyIOException(ILogger logger, IOException exception)
+            private static readonly Action<ILogger, string, string, string, Exception?> _parameterBindingFailed = LoggerMessage.Define<string, string, string>(
+                LogLevel.Debug,
+                new EventId(3, "ParamaterBindingFailed"),
+                @"Failed to bind parameter ""{ParameterType} {ParameterName}"" from ""{SourceValue}"".");
+
+            public static void RequestBodyIOException(HttpContext httpContext, IOException exception)
+            {
+                _requestBodyIOException(GetLogger(httpContext), exception);
+            }
+
+            public static void RequestBodyInvalidDataException(HttpContext httpContext, InvalidDataException exception)
+            {
+                _requestBodyInvalidDataException(GetLogger(httpContext), exception);
+            }
+
+            public static void ParameterBindingFailed(HttpContext httpContext, string parameterTypeName, string parameterName, string sourceValue)
             {
-                _requestBodyIOException(logger, exception);
+                _parameterBindingFailed(GetLogger(httpContext), parameterTypeName, parameterName, sourceValue, null);
             }
 
-            public static void RequestBodyInvalidDataException(ILogger logger, InvalidDataException exception)
+            private static ILogger GetLogger(HttpContext httpContext)
             {
-                _requestBodyInvalidDataException(logger, exception);
+                var loggerFactory = httpContext.RequestServices.GetRequiredService<ILoggerFactory>();
```

Comment:
Capture the logger once, this isn't free.

Reply:
Meh, not important for this change but we'll need to optimize

Reply:
This optimizes the failure case quite a bit by not relying on exceptions. I think multiple parse failures are rare enough we can avoid optimizing it to that degree.